tests/integration/questionnaire/test_questionnaire_list_collector_repeating_blocks.py

```diff
@@ -474,3 +474,25 @@ def test_edit_repeating_block_from_submit_page_returns_to_submit_page(
         # previous link should return to submit page
         self.previous()
         self.assertInUrl("/submit/")
+
+    def test_previous_link_in_two_list_collectors_with_repeating_blocks_returns_to_previous_location(
+        self,
+    ):
+        self.launchSurvey("test_list_collector_repeating_blocks_with_hub")
+        self.post({"responsible-party-answer": "Yes"})
+
+        # Add some items and progress to second list collector
+        self.add_three_companies()
+        self.post({"any-other-companies-or-branches-answer": "No"})
+        self.post({"any-other-trading-details-answer": "No other details"})
+        self.post()
+        self.get("/questionnaire/sections/section-businesses/")
```

Comment:
This tests needs more well-defined comments, it is not obvious why it is doing this in this way. i.e when coming back to it, it is not clear why we just randomly go to some specific URL. I suggest adding comments around why it means, i.e this is the second list collector section etc.

Reply:
Added a lot of descriptions so the journey should be clear, changed that specific URL to just another `post` so it's clearer as well.